app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/importlogins/ImportLoginsViewModelTest.kt

```diff
@@ -207,15 +228,107 @@ class ImportLoginsViewModelTest : BaseViewModelTest() {
             ImportLoginsState(
                 dialogState = null,
                 viewState = ImportLoginsState.ViewState.InitialContent,
+                isVaultSyncing = false,
+            ),
+            viewModel.stateFlow.value,
+        )
+    }
+
+    @Test
+    fun `MoveToSyncInProgress sets isVaultSyncing to true and calls syncForResult`() {
+        val viewModel = createViewModel()
+        viewModel.trySendAction(ImportLoginsAction.MoveToSyncInProgress)
+        assertEquals(
+            ImportLoginsState(
+                dialogState = null,
+                viewState = ImportLoginsState.ViewState.InitialContent,
+                isVaultSyncing = true,
             ),
             viewModel.stateFlow.value,
         )
+        coVerify { vaultRepository.syncForResult() }
     }
 
-    private fun createViewModel(): ImportLoginsViewModel = ImportLoginsViewModel()
+    @Suppress("MaxLineLength")
+    @Test
+    fun `RetryVaultSync sets isVaultSyncing to true and clears dialog state and calls syncForResult`() =
+        runTest {
+            coEvery { vaultRepository.syncForResult() } returns SyncVaultDataResult.Error(Exception())
+            val viewModel = createViewModel()
+            viewModel.trySendAction(ImportLoginsAction.MoveToSyncInProgress)
+            viewModel.stateFlow.test {
+                assertNotNull(awaitItem().dialogState)
+                coEvery { vaultRepository.syncForResult() } returns SyncVaultDataResult.Success
+                viewModel.trySendAction(ImportLoginsAction.RetryVaultSync)
+                assertEquals(
+                    ImportLoginsState(
+                        dialogState = null,
+                        viewState = ImportLoginsState.ViewState.InitialContent,
+                        isVaultSyncing = true,
+                    ),
+                    awaitItem(),
+                )
+            }
+            coVerify { vaultRepository.syncForResult() }
+        }
+
+    @Test
+    fun `When sync succeeds should send NavigateToImportSuccess event`() = runTest {
+        val viewModel = createViewModel()
+        viewModel.eventFlow.test {
+            viewModel.trySendAction(ImportLoginsAction.MoveToSyncInProgress)
+            assertEquals(ImportLoginsEvent.NavigateToImportSuccess, awaitItem())
+        }
+    }
+
+    @Test
+    fun `When sync fails should remove loading state and show error dialog`() = runTest {
```

Comment:
:pick: Similar to above, `SyncVaultDataResult Error should remove loading state and show error dialog`, is closer to how we've been naming tests.